core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -57,7 +57,10 @@ class AdminClient(val time: Time,
       now = time.milliseconds()
     } while (now < deadline && future.exception().isInstanceOf[SendFailedException])
 
-    throw future.exception()
+    if (future.failed())
+      throw future.exception()
+    else
+      throw new TimeoutException();
```

Comment:
Sorry that I missed this earlier, but it would be good to include a message in this exception.